### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by darylmartin100**
### Issues:
Resolves CryptoAlg-542

### Description of changes: 
Fixing a reference to s2n, changing references of aws-lc to AWS-LC, and adding instructions on how to Fork the repo as part of our quick start guide.

### Call-outs:
N/A
### Testing:
Performed a preview of the document in github.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,5 +48,5 @@ If you have any questions about Submitting PR's, Opening Issues, aws-lc API usag
 Otherwise, if you think you might have found a security impacting issue, please instead follow [our Security Notification Process.](#security-issue-notifications)
 
 ## Security issue notifications
-If you discover a potential security issue in s2n we ask that you notify
+If you discover a potential security issue in aws-lc we ask that you notify
```
Comment (torben-hansen) [inline README.md:51]: AWS-LC?

**Review by torben-hansen — COMMENTED** [review]
Any other aws-lc that should be AWS-LC instead?

File: README.md [hunk 2]
```diff
@@ -19,7 +19,16 @@ export PATH="$GOROOT/bin:$PATH"
...
 ```
-git clone https://github.com/${YOUR_GITHUB_ACCOUNT_NAME}/aws-lc.git
+# Create your fork in GitHub if it doesn't exist and clone it
+git clone git@github.com:${GITHUB_USERNAME}/aws-lc.git && cd aws-lc
+git remote add upstream git@github.com:awslabs/aws-lc.git
+
+# Update your fork and branch if needed
+git checkout main
+git fetch upstream
+git merge upstream/main
```
Comment (nebeid) [inline README.md:29]: I do rebase in this step instead of merge. I don't know if it's worth mentioning.

File: README.md [hunk 3]
```diff
@@ -19,7 +19,16 @@ export PATH="$GOROOT/bin:$PATH"
...
+git push origin/main
```
Comment (nebeid) [inline README.md:30]: I have needed push -f

File: README.md [hunk 4]
```diff
@@ -39,5 +48,5 @@ If you have any questions about Submitting PR's, Opening Issues, aws-lc API usag
 Otherwise, if you think you might have found a security impacting issue, please instead follow [our Security Notification Process.](#security-issue-notifications)
```
Comment (nebeid) [inline README.md:48]: Is this Gitter chatroom now public? If so, may be comment on the issue?

**Review by torben-hansen — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (darylmartin100) [reply README.md:48]: Yup, it's public now.

**Pull request merged**